llvm/lib/CAS/LazyMappedFileRegion.cpp

```diff
@@ -242,14 +242,60 @@ Expected<LazyMappedFileRegion> LazyMappedFileRegion::create(
 
   sys::fs::file_t File = sys::fs::convertFDToNativeFile(FD);
 
+  struct FileLockRAII {
```

Comment:
There is already `FileLocker` in FileSystem.h; could  we extend that to take LockKind?  It seems like the pattern there is `raw_fs_ostream::lock()` returns a `FileLocker` that is already locked; maybe we could have `LazyMappedFileRegion::lock()` to match that?

Reply:
`FileLocker` is only usable from `raw_fd_ostream`. I could see potential benefit to investigate making `FileLocker` more broadly useful but I'd prefer it as follow-up refactoring.